Trilinos: Add STRUMPACK dependency

pghysels opened the pull request:
The STRUMPACK interface was added to Trilinos in this PR: https://github.com/trilinos/Trilinos/pull/8192
See also this issue: https://github.com/trilinos/Trilinos/issues/7413

I found it strange that I had to add: 
`define('Amesos2_ENABLE_STRUMPACK', True)`
even though `TPL_ENABLE_STRUMPACK=ON` is already set.

@keitat @jwillenbring @ndellingwood @srajama1 
@lisaclaus @xiaoyeli

pghysels committed:
Trilinos: Add STRUMPACK dependency

pghysels committed:
break long lines, flake8 cleanup

adamjstewart commented on var/spack/repos/builtin/packages/trilinos/package.py:
```diff
@@ -650,6 +656,17 @@ def define_tpl_enable(cmake_var, spec_var=None):
+                define('TPL_ENABLE_STRUMPACK', True),
+                define('Amesos2_ENABLE_STRUMPACK', True),
+                define('STRUMPACK_LIBRARY_DIRS',
+                       spec['strumpack'].prefix.lib),
```
```suggestion
                       spec['strumpack'].libs.directories[0]),
```
Libraries may be in `lib` or `lib64`

adamjstewart commented on var/spack/repos/builtin/packages/trilinos/package.py:
```diff
@@ -650,6 +656,17 @@ def define_tpl_enable(cmake_var, spec_var=None):
+                define('STRUMPACK_LIBRARY_DIRS',
+                       spec['strumpack'].prefix.lib),
+                define('STRUMPACK_INCLUDE_DIRS',
+                       spec['strumpack'].prefix.include),
```
```suggestion
                       spec['strumpack'].headers.directories[0]),
```

pghysels committed:
Use spec['strumpack'].libs.directories[0]
instead of spec['strumpack'].prefix.lib
because libraries may be in lib or lib64.

Likewise use headers.directories[0] iso prefix.include.

Suggested by adamjstewart
